src/core/unit.c

```diff
@@ -4162,15 +4162,22 @@ UnitFileState unit_get_unit_file_state(Unit *u) {
 
         assert(u);
 
-        if (u->unit_file_state < 0 && u->fragment_path) {
-                r = unit_file_get_state(
-                                u->manager->runtime_scope,
-                                NULL,
-                                u->id,
-                                &u->unit_file_state);
-                if (r < 0)
-                        u->unit_file_state = UNIT_FILE_BAD;
-        }
+        if (u->unit_file_state >= 0 || !u->fragment_path)
+                return u->unit_file_state;
+
+        if (u->transient)
+                return (u->unit_file_state = UNIT_FILE_TRANSIENT);
+
+        if (u->perpetual)
+                return (u->unit_file_state = UNIT_FILE_STATIC);
```

Comment:
This seems wrong? E.g. for -.mount whether it's generated or not does make a difference?

Reply:
indeed